File: gluecodium/src/test/resources/smoke/inheritance/output/android/com/example/foobar/CrossPackageChildInterfaceImpl.java
/*
 *
 */
package com.example.foobar;
import android.support.annotation.NonNull;
import com.example.NativeBase;
class CrossPackageChildInterfaceImpl extends NativeBase implements CrossPackageChildInterface {
    protected CrossPackageChildInterfaceImpl(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @Override
    public native void rootMethod();
    @NonNull
    @Override
    public native String getRootProperty();
    @Override
    public native void setRootProperty(@NonNull final String value);
}

File: gluecodium/src/test/resources/smoke/inheritance/output/android-kotlin/com/example/foobar/CrossPackageChildInterfaceImpl.kt
/*

 *
 */

@file:JvmName("CrossPackageChildInterfaceImpl")

package com.example.foobar

import com.example.NativeBase

/**
 * @suppress
 */
class CrossPackageChildInterfaceImpl : NativeBase, CrossPackageChildInterface {
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    override external fun rootMethod() : Unit
    override var rootProperty: String
        external get
        external set


    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
